whois-nrtm4/src/main/java/net/ripe/db/nrtm4/persist/WhoisSlaveRepository.java

```diff
@@ -0,0 +1,61 @@
+package net.ripe.db.nrtm4.persist;
+
+import net.ripe.db.whois.common.dao.jdbc.domain.ObjectTypeIds;
+import net.ripe.db.whois.common.domain.serials.Operation;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.javatuples.Pair;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.List;
+
+
+@Repository
+public class WhoisSlaveRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    final String sqlSerialAndRpslTemplate = "" +
+        "SELECT " +
+        "s.serial_id, s.object_id, s.sequence_id, s.atlast, s.operation, " +
+        "r.object_id, r.sequence_id, r.timestamp, r.object_type, r.object, r.pkey " +
+        "FROM serials s JOIN %s r ON r.object_id = s.object_id and r.sequence_id = s.sequence_id " +
+        "WHERE s.serial_id > ? " +
+        "ORDER BY s.serial_id ASC";
+
+    private final RowMapper<Pair<SerialModel, RpslObjectModel>> serialAndRpslObjectMapper = (rs, n) ->
+        new Pair<>(
+            new SerialModel(
+                rs.getInt(1),
```

Comment:
We already have a bean name VersionInfo bean for the same (SerialModel) in whois-commons, can you have a look at the  existing code? May be refactor or shift this code to versionDao instead?

Reply:
it doesn't have the serial_id column, which we need for ordering the changes. maybe it's possible to extend so i can add the extra column. but there's sth odd about that class -- it implements `identifiable` on the object_id column but that table is not unique on object_id. let's discuss it.

Reply:
as discussed -- it doesn't have to be unique to be 'Identifiable' (although it's used in the code like it is!). adding serial_id to the SELECT doesn't have any effect anyway.